setup.py

```diff
@@ -24,7 +24,10 @@
 requirements = [
     "qiskit>=0.19.1",
     "pennylane>=0.9.0",
-    "numpy"
+    "numpy",
+    "networkx>=2.2;python_version>'3.5'",
+    # Networkx 2.4 is the final version with python 3.5 support.
+    "networkx>=2.2,<2.4;python_version=='3.5'",
```

Comment:
@antalszava, why do we need to pin a downstream dependency? When the user pip installs `pennylane-qiskit`, `qiskit` will be installed first, where this pinning is enforced.

Reply:
Typically, we don't specify downstream dependencies in packages if they are not an explicit dependency of the package

Reply:
When installing the wheel in a Python 3.5 environment (e.g. note that the collected packages have `cp35` tags), this was not the case locally, unless the `setup.py` was updated.
```
Collecting networkx
  Using cached networkx-2.4-py3-none-any.whl (1.6 MB)
```
Resulting in
```python
ERROR: qiskit-terra 0.14.1 has requirement networkx<2.4,>=2.2; python_version == "3.5", but you'll have networkx 2.4 which is incompatible.
```
**Entire trace:**
```python
pip install PennyLane_qiskit-0.9.0-py3-none-any.whl
Processing ./PennyLane_qiskit-0.9.0-py3-none-any.whl
Collecting pennylane>=0.9.0
  Using cached PennyLane-0.9.0-py3-none-any.whl (239 kB)
Collecting qiskit>=0.19.1
  Using cached qiskit-0.19.2.tar.gz (3.9 kB)
Collecting numpy
  Using cached numpy-1.18.4-cp35-cp35m-manylinux1_x86_64.whl (20.0 MB)
Processing /.cache/pip/wheels/42/62/66/1121afe23ff96af4e452e0d15e68761e3f605952ee075ca99f/autograd-1.3-cp35-none-any.whl
Collecting scipy
  Using cached scipy-1.4.1-cp35-cp35m-manylinux1_x86_64.whl (26.0 MB)
Collecting semantic-version==2.6
  Using cached semantic_version-2.6.0-py3-none-any.whl (14 kB)
Collecting networkx
  Using cached networkx-2.4-py3-none-any.whl (1.6 MB)
Collecting toml
  Using cached toml-0.10.1-py2.py3-none-any.whl (19 kB)
Collecting appdirs
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting qiskit-terra==0.14.1
  Using cached qiskit_terra-0.14.1-cp35-cp35m-manylinux2010_x86_64.whl (6.7 MB)
Collecting qiskit-aer==0.5.1
  Using cached qiskit_aer-0.5.1-cp35-cp35m-manylinux2010_x86_64.whl (23.3 MB)
Collecting qiskit-ibmq-provider==0.7.1
  Downloading qiskit_ibmq_provider-0.7.1-py3-none-any.whl (155 kB)
     |████████████████████████████████| 155 kB 2.8 MB/s 
Collecting qiskit-ignis==0.3.0
  Using cached qiskit_ignis-0.3.0-py3-none-any.whl (172 kB)
Collecting qiskit-aqua==0.7.1
  Downloading qiskit_aqua-0.7.1-py3-none-any.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB 3.2 MB/s 
Processing /.cache/pip/wheels/8b/99/a0/81daf51dcd359a9377b110a8a886b3895921802d2fc1b2397e/future-0.18.2-cp35-none-any.whl
Collecting decorator>=4.3.0
  Using cached decorator-4.4.2-py2.py3-none-any.whl (9.2 kB)
Collecting marshmallow<4,>=3
  Using cached marshmallow-3.6.0-py2.py3-none-any.whl (45 kB)
Processing /.cache/pip/wheels/34/31/15/7b070b25d0a549d20ce2e9fe6d727471c2c61ef904720fd40c/python_constraint-1.4.0-py2.py3-none-any.whl
Collecting ply>=3.10
  Using cached ply-3.11-py2.py3-none-any.whl (49 kB)
Collecting fastjsonschema>=2.10
  Using cached fastjsonschema-2.14.4-py3-none-any.whl (21 kB)
Processing /.cache/pip/wheels/59/b1/91/f02e76c732915c4015ab4010f3015469866c1eb9b14058d8e7/dill-0.3.1.1-cp35-none-any.whl
Collecting python-dateutil>=2.8.0
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Processing /.cache/pip/wheels/d7/69/b4/3200b95828d1f0ddb3cb5699083717f4fdbd9b4223d0644c57/psutil-5.7.0-cp35-cp35m-linux_x86_64.whl
Collecting sympy>=1.3
  Using cached sympy-1.5.1-py2.py3-none-any.whl (5.6 MB)
Processing /.cache/pip/wheels/69/27/3e/7472f75eab2db111aa387715fe009af15899459517fc1a1906/marshmallow_polyfield-5.9-cp35-none-any.whl
Collecting jsonschema>=2.6
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting retworkx>=0.3.2
  Using cached retworkx-0.3.4-cp35-cp35m-manylinux2010_x86_64.whl (762 kB)
Collecting cython>=0.27.1
  Using cached Cython-0.29.17-cp35-cp35m-manylinux1_x86_64.whl (2.1 MB)
Requirement already satisfied: pybind11>=2.4 in /anaconda3/envs/pl_qiskit_3_5/lib/python3.5/site-packages (from qiskit-aer==0.5.1->qiskit>=0.19.1->PennyLane-qiskit==0.9.0) (2.5.0)
Collecting urllib3>=1.21.1
  Using cached urllib3-1.25.9-py2.py3-none-any.whl (126 kB)
Collecting requests-ntlm>=1.1.0
  Using cached requests_ntlm-1.1.0-py2.py3-none-any.whl (5.7 kB)
Collecting websockets<8,>=7
  Using cached websockets-7.0-cp35-cp35m-manylinux1_x86_64.whl (63 kB)
Collecting nest-asyncio!=1.1.0,>=1.0.0
  Using cached nest_asyncio-1.3.3-py3-none-any.whl (4.7 kB)
Collecting requests>=2.19
  Using cached requests-2.23.0-py2.py3-none-any.whl (58 kB)
Requirement already satisfied: setuptools>=40.1.0 in /anaconda3/envs/pl_qiskit_3_5/lib/python3.5/site-packages (from qiskit-ignis==0.3.0->qiskit>=0.19.1->PennyLane-qiskit==0.9.0) (40.2.0)
Collecting h5py
  Using cached h5py-2.10.0-cp35-cp35m-manylinux1_x86_64.whl (2.8 MB)
Collecting pyscf; sys_platform != "win32"
  Using cached pyscf-1.7.2-cp35-cp35m-manylinux1_x86_64.whl (26.0 MB)
Processing /.cache/pip/wheels/aa/28/cd/ba75f92d49b25dacec6b80f99efcc393dc84abf33f96632ac4/docplex-2.14.186-cp35-none-any.whl
Collecting scikit-learn>=0.20.0
  Using cached scikit_learn-0.22.2.post1-cp35-cp35m-manylinux1_x86_64.whl (7.0 MB)
Processing /.cache/pip/wheels/bb/ba/15/fdd0deb104df3254912998150ba9245668db06b00af5912d1a/dlx-1.0.4-cp35-none-any.whl
Processing /.cache/pip/wheels/7e/95/df/935f19af51d6ab63d416c103c0d6eae7048a8c2ecb00a1497f/fastdtw-0.3.4-cp35-cp35m-linux_x86_64.whl
Collecting quandl
  Using cached Quandl-3.5.0-py2.py3-none-any.whl (25 kB)
Requirement already satisfied: six>=1.5 in /anaconda3/envs/pl_qiskit_3_5/lib/python3.5/site-packages (from python-dateutil>=2.8.0->qiskit-terra==0.14.1->qiskit>=0.19.1->PennyLane-qiskit==0.9.0) (1.14.0)
Processing /.cache/pip/wheels/63/9d/8e/37c3f6506ed3f152733a699e92d8e0c9f5e5f01dea262f80ad/mpmath-1.1.0-cp35-none-any.whl
Processing /.cache/pip/wheels/c2/85/ad/bc6d41e2c4b35c9fdfed48f0fcd411ffc4164e67755ddf9ebb/pyrsistent-0.16.0-cp35-cp35m-linux_x86_64.whl
Collecting attrs>=17.4.0
  Using cached attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Collecting importlib-metadata; python_version < "3.8"
  Using cached importlib_metadata-1.6.0-py2.py3-none-any.whl (30 kB)
Collecting ntlm-auth>=1.0.2
  Using cached ntlm_auth-1.4.0-py2.py3-none-any.whl (29 kB)
Collecting cryptography>=1.3
  Using cached cryptography-2.9.2-cp35-abi3-manylinux2010_x86_64.whl (2.7 MB)
Collecting chardet<4,>=3.0.2
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting idna<3,>=2.5
  Using cached idna-2.9-py2.py3-none-any.whl (58 kB)
Requirement already satisfied: certifi>=2017.4.17 in /anaconda3/envs/pl_qiskit_3_5/lib/python3.5/site-packages (from requests>=2.19->qiskit-ibmq-provider==0.7.1->qiskit>=0.19.1->PennyLane-qiskit==0.9.0) (2018.8.24)
Collecting joblib>=0.11
  Using cached joblib-0.14.1-py2.py3-none-any.whl (294 kB)
Collecting inflection>=0.3.1
  Using cached inflection-0.4.0-py2.py3-none-any.whl (5.8 kB)
Collecting more-itertools
  Using cached more_itertools-8.2.0-py3-none-any.whl (43 kB)
Collecting pandas>=0.14
  Using cached pandas-0.25.3-cp35-cp35m-manylinux1_x86_64.whl (10.3 MB)
Collecting zipp>=0.5
  Using cached zipp-1.2.0-py2.py3-none-any.whl (4.8 kB)
Collecting cffi!=1.11.3,>=1.8
  Using cached cffi-1.14.0-cp35-cp35m-manylinux1_x86_64.whl (399 kB)
Collecting pytz>=2017.2
  Using cached pytz-2020.1-py2.py3-none-any.whl (510 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Building wheels for collected packages: qiskit
  Building wheel for qiskit (setup.py) ... done
  Created wheel for qiskit: filename=qiskit-0.19.2-py3-none-any.whl size=2899 sha256=abcd222bac19b816c9a9955d23cc16b40a826c90c1139c603654c29f480260bb
  Stored in directory: /.cache/pip/wheels/61/f5/ae/206bfcd20eddeae3b1c527b0ff2ec669d313e09061e442262e
Successfully built qiskit
ERROR: qiskit-terra 0.14.1 has requirement networkx<2.4,>=2.2; python_version == "3.5", but you'll have networkx 2.4 which is incompatible.
Installing collected packages: numpy, future, autograd, scipy, semantic-version, decorator, networkx, toml, appdirs, pennylane, marshmallow, python-constraint, ply, fastjsonschema, dill, python-dateutil, psutil, mpmath, sympy, marshmallow-polyfield, pyrsistent, attrs, zipp, importlib-metadata, jsonschema, retworkx, qiskit-terra, cython, qiskit-aer, urllib3, chardet, idna, requests, ntlm-auth, pycparser, cffi, cryptography, requests-ntlm, websockets, nest-asyncio, qiskit-ibmq-provider, qiskit-ignis, h5py, pyscf, docplex, joblib, scikit-learn, dlx, fastdtw, inflection, more-itertools, pytz, pandas, quandl, qiskit-aqua, qiskit, PennyLane-qiskit
Successfully installed PennyLane-qiskit-0.9.0 appdirs-1.4.4 attrs-19.3.0 autograd-1.3 cffi-1.14.0 chardet-3.0.4 cryptography-2.9.2 cython-0.29.17 decorator-4.4.2 dill-0.3.1.1 dlx-1.0.4 docplex-2.14.186 fastdtw-0.3.4 fastjsonschema-2.14.4 future-0.18.2 h5py-2.10.0 idna-2.9 importlib-metadata-1.6.0 inflection-0.4.0 joblib-0.14.1 jsonschema-3.2.0 marshmallow-3.6.0 marshmallow-polyfield-5.9 more-itertools-8.2.0 mpmath-1.1.0 nest-asyncio-1.3.3 networkx-2.4 ntlm-auth-1.4.0 numpy-1.18.4 pandas-0.25.3 pennylane-0.9.0 ply-3.11 psutil-5.7.0 pycparser-2.20 pyrsistent-0.16.0 pyscf-1.7.2 python-constraint-1.4.0 python-dateutil-2.8.1 pytz-2020.1 qiskit-0.19.2 qiskit-aer-0.5.1 qiskit-aqua-0.7.1 qiskit-ibmq-provider-0.7.1 qiskit-ignis-0.3.0 qiskit-terra-0.14.1 quandl-3.5.0 requests-2.23.0 requests-ntlm-1.1.0 retworkx-0.3.4 scikit-learn-0.22.2.post1 scipy-1.4.1 semantic-version-2.6.0 sympy-1.5.1 toml-0.10.1 urllib3-1.25.9 websockets-7.0 zipp-1.2.0
```

Reply:
Ah, I think this is an odd ordering issue 🤔 My guess is that 
```
pip install pennylane
pip install pennylane-qiskit
```
should avoid this issue. Still, this is an annoying 'bug'!

Reply:
The issue seems to come from the fact that the PL dependencies are processed first _before_ the qiskit dependencies. Pip likely just does it in alphabetical order.

Reply:
oh `pip`... :crying_cat_face: